File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/boudarycondition/BoundaryCondition.scala
package pl.edu.agh.multiscalemodelling.engine.logic.boudarycondition

import pl.edu.agh.multiscalemodelling.engine.logic.Point

abstract class BoundaryCondition {
  def getPosition(position: Point, size: Point): Point

  def skip(position: Point, size: Point): Boolean
}

File: core/src/main/scala/pl/edu/agh/multiscalemodelling/engine/logic/boudarycondition/BoundaryCondition.java
package pl.edu.agh.multiscalemodelling.engine.logic.boudarycondition;

import pl.edu.agh.multiscalemodelling.engine.logic.Point;

public abstract class BoundaryCondition {

    public abstract Point getPosition(Point position, Point size);

    public abstract boolean skip(Point position, Point size);

}
